tests/test_git_colocated.rs

```diff
@@ -235,3 +235,104 @@ fn test_git_colocated_conflicting_git_refs() {
+    std::fs::write(workspace_root.join("file"), "contents").unwrap();
+    git_repo
+        .index()
+        .unwrap()
+        .add_path(Path::new("file"))
+        .unwrap();
+    let tree1_oid = git_repo.index().unwrap().write_tree().unwrap();
+    let tree1 = git_repo.find_tree(tree1_oid).unwrap();
+    let signature = git2::Signature::new(
+        "Someone",
+        "someone@example.com",
+        &git2::Time::new(1234567890, 60),
+    )
+    .unwrap();
+    let oid1 = git_repo
+        .commit(
+            Some("refs/heads/master"),
+            &signature,
+            &signature,
+            "initial",
+            &tree1,
+            &[],
+        )
+        .unwrap();
+    insta::assert_snapshot!(
+        git_repo.head().unwrap().peel_to_commit().unwrap().id().to_string(),
+        @"e61b6729ff4292870702f2f72b2a60165679ef37"
+    );
+
+    // Add a second commit in Git
+    std::fs::write(workspace_root.join("file"), "other contents").unwrap();
+    git_repo
+        .index()
+        .unwrap()
+        .add_path(Path::new("file"))
+        .unwrap();
+    let tree2_oid = git_repo.index().unwrap().write_tree().unwrap();
+    let tree2 = git_repo.find_tree(tree2_oid).unwrap();
+    let signature = git2::Signature::new(
+        "Someone",
+        "someone@example.com",
+        &git2::Time::new(1234567890, 62),
+    )
+    .unwrap();
+    let oid2 = git_repo
+        .commit(
+            Some("refs/heads/master"),
+            &signature,
+            &signature,
+            "next",
+            &tree2,
+            &[&git_repo.find_commit(oid1).unwrap()],
+        )
+        .unwrap();
+    insta::assert_snapshot!(
+        git_repo.head().unwrap().peel_to_commit().unwrap().id().to_string(),
+        @"9348a77eb52cf85cdc3aedb6b5279bc1727ace4c"
+    );
+
+    // Remove the second commit in Git
+    let mut checkout_builder = git2::build::CheckoutBuilder::new();
+    git_repo
+        .reset(
+            &git_repo
+                .find_object(oid1, Some(git2::ObjectType::Commit))
+                .unwrap(),
+            git2::ResetType::Hard,
+            Some(&mut checkout_builder),
+        )
+        .unwrap();
+    insta::assert_snapshot!(
+        git_repo.head().unwrap().peel_to_commit().unwrap().id().to_string(),
+        @"e61b6729ff4292870702f2f72b2a60165679ef37"
+    );
+
+    // Import the repo while there is no path to the second commit
+    test_env.jj_cmd_success(&workspace_root, &["init", "--git-repo", "."]);
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_root), @r###"
+    @ 3e9369cd54227eb88455e1834dbc08aad6a16ac4 
+    o e61b6729ff4292870702f2f72b2a60165679ef37 master
+    o 0000000000000000000000000000000000000000 
+    "###);
+    insta::assert_snapshot!(
+        git_repo.head().unwrap().peel_to_commit().unwrap().id().to_string(),
+        @"e61b6729ff4292870702f2f72b2a60165679ef37"
+    );
+
+    // Check that the second commit cannot be found by jj without crashing
```

Comment:
nit: "cannot be found without crashing" makes it sound like we can find it if we crash :) Maybe something like "Check that trying to look up the second commit fails gracefully"

Reply:
Done.